Managers/FrameMan.cpp

```diff
@@ -1,2651 +1,1391 @@
+
+			//Draw sky gradient
+			for (int i = 0; i < pWorldBitmap->h; i++) {
+				hline(pWorldBitmap, 0, i, pWorldBitmap->w - 1, makecol32(64 + (((float)i / (float)pWorldBitmap->h) * (128 - 64)), 64 + (((float)i / (float)pWorldBitmap->h) * (192 - 64)), 96 + ((float)i / (float)pWorldBitmap->h) * (255 - 96)));
+			}
+
+			// Draw scene
+			draw_sprite(pWorldBitmap, g_SceneMan.GetTerrain()->GetBGColorBitmap(), 0, 0);
+			draw_sprite(pWorldBitmap, g_SceneMan.GetTerrain()->GetFGColorBitmap(), 0, 0);
+
+			//Draw objects
+			draw_sprite(pWorldBitmap, g_SceneMan.GetMOColorBitmap(), 0, 0);
+
+			g_PostProcessMan.GetPostScreenEffectsWrapped(targetPos, pWorldBitmap->w, pWorldBitmap->h, postEffects, -1);
+
+			//Draw post-effects
+			BITMAP *pBitmap = 0;
+			int effectPosX = 0;
+			int effectPosY = 0;
+			int strength = 0;
+			float angle = 0;
+
+			for (list<PostEffect>::iterator eItr = postEffects.begin(); eItr != postEffects.end(); ++eItr) {
+				pBitmap = (*eItr).m_pBitmap;
+				strength = (*eItr).m_Strength;
+				set_screen_blender(strength, strength, strength, strength);
+				effectPosX = (*eItr).m_Pos.GetFloorIntX() - (pBitmap->w / 2);
+				effectPosY = (*eItr).m_Pos.GetFloorIntY() - (pBitmap->h / 2);
+				angle = (*eItr).m_Angle;
+
+				if (angle == 0) {
+					draw_trans_sprite(pWorldBitmap, pBitmap, effectPosX, effectPosY);
+				} else {
+					BITMAP * pTargetBitmap;
+
+					if (pBitmap->w < 16 && pBitmap->h < 16) {
+						pTargetBitmap = g_PostProcessMan.GetTempEffectBitmap(16);
+					} else if (pBitmap->w < 32 && pBitmap->h < 32) {
+						pTargetBitmap = g_PostProcessMan.GetTempEffectBitmap(32);
+					} else if (pBitmap->w < 64 && pBitmap->h < 64) {
+						pTargetBitmap = g_PostProcessMan.GetTempEffectBitmap(64);
+					} else if (pBitmap->w < 128 && pBitmap->h < 128) {
+						pTargetBitmap = g_PostProcessMan.GetTempEffectBitmap(128);
+					} else if (pBitmap->w < 256 && pBitmap->h < 256) {
+						pTargetBitmap = g_PostProcessMan.GetTempEffectBitmap(256);
+					} else {
+						pTargetBitmap = g_PostProcessMan.GetTempEffectBitmap(512);
+					}
+					clear_to_color(pTargetBitmap, 0);
+
+					fixed fAngle;
+					fAngle = fixmul(angle, radtofix_r);
+
+					rotate_sprite(pTargetBitmap, pBitmap, 0, 0, fAngle);
+					draw_trans_sprite(pWorldBitmap, pTargetBitmap, effectPosX, effectPosY);
+				}
+			}
+
+			PALETTE palette;
+			get_palette(palette);
+			save_bmp(fullfilename, pWorldBitmap, palette);
+
+			g_ConsoleMan.PrintString("SYSTEM: World was dumped to: " + string(fullfilename));
+
+			destroy_bitmap(pWorldBitmap);
+		}
+		return 0;
 	}
-	return m_PlayerScreenHeight; 
-}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int FrameMan::SaveBitmapToBMP(BITMAP *pBitmap, const char *namebase) {
+		int filenumber = 0;
+		char fullfilename[256];
+		int maxFileTrys = 1000;
 
-} // namespace RTE
+		// Make sure its not a 0 name base
```

Comment:
junk comment.

Also the do-while below.

Also also, given how similar this and SaveScreenToBMP are, it'd be nice if SaveScreenToBMP just did a little bit of safety checking and called this.
i.e. something like
```
if (screen) {
	if (!m_pScreendumpBuffer) { m_pScreendumpBuffer = create_bitmap(screen->w, screen->h); }
	blit(screen, m_pScreendumpBuffer, 0, 0, 0, 0, screen->w, screen->h);

	return SaveBitmapToBMP(m_pScreendumpBuffer, namebase);
}
return 0;
```

You could probably also do that for SaveWorldToBMP, basically keeping all the code there that's different, then doing `return SaveBitmapToBMP(pWorldBitmap, namebase);`

I'd say keep the check for a valid name base though, to save on effort if there's an invalid name base passed in.
```

if (namebase == 0 || strlen(namebase) <= 0) {
	return -1;
}
```